### DISCUSSION TITLE
configuration via command line after 5.4 release

### DISCUSSION THREAD

**Discussion by ghost** [jcorporation/myMPD#129] [Q&A] [answered]
Before version 5.4 I was able to configure everything via the mympd.conf file. With 5.4, mympd.conf is not used for most of mympd settings so I decided to do some tests with /var/lib/mympd/state files.

I need to change background, enable cover images, name the cover Folder.jpg and set MPD music directory, so I did the following:

1. I stopped mympd service:

`systemctl stop mympd`

2. I gave the following commands:
```
echo "#FFF" >  /var/lib/mympd/state/bg_color
echo "true" > /var/lib/mympd/state/coverimage
echo "Folder.jpg" > /var/lib/mympd/state/coverimage_name
echo "/var/lib/mpd/music" > /var/lib/mympd/state/music_directory
```
3. and finally fixed ownership and permissions:

```
chown mympd:mympd /var/lib/mympd/state/*
chmod 600 /var/lib/mympd/state/*
```

4. started mympd

`systemctl start mympd`

Apart from the new background color, none of the other commands did the job.

I had a look at the web interface settings, and realized that:

- music directory was not set in **MPD Connection**
- **Albumart** was not enabled

Am I missing something?

The reason I am trying to configure this way, Is because I want to create an Archlinux package with all my settings pre-configured, so all these echo commands, will hopefully be placed withing my PKGBUILD.

Thanks in advance.

**Comment by jcorporation** [comment, marked as answer]
All configuration can be done through mympd.conf. The initial values of the state files are from mympd.conf environment or compile time defaults. See the wiki under https://github.com/jcorporation/myMPD/wiki/Configuration 

Also your approach should work. Your error is that you write a newline character in the state files, use echo -n and it should work.

Configuration through mympd.conf is my prefered way. Ship a empty state dir.

**Comment by ghost** [comment]
Thanks for the reply.

I kept using some old variables in mympd.conf file and this is why they did not work. 

I updated as per the new sections and names and now it's ok.
